### DISCUSSION TITLE
Get an object from string?

### DISCUSSION THREAD

**Discussion by avidan-efody** [cocotb/cocotb#2599] [Q&A]
I have python code that reads fsdb/simv SNPS databases and extracts signal names as strings.

I think I can assign signals using cocotb like this: `dut.cluster.block.sub_block.in.Force(1)`

How can I assign a signal if I have the name as string, i.e. `"dut.cluster.block.sub_block.in"`?

Of course I can use python 'exec()', but wondering if there's some API to go from string to object that I'm missing?

Thanks!

**Comment by eric-wieser** [comment]
I don't think the API is builtin anyway, and if it is it's definitely not public; but instead of using `exec` you can use `getattr`:
```python
s = "dut.cluster.block.sub_block.in"
parts = s.split('.')

# it might make sense to not include `dut` in the string in the first place,
# since your config strings shouldn't depend on your python variable names.
assert parts[0] == 'dut'
sig = dut

# step through all the other parts using `getattr`
for p in parts[1:]:
    sig = getattr(sig, p)
```
or even more concisely:
```python
from functools import reduce

s = "cluster.block.sub_block.in"  # note: no `dut`

sig = reduce(getattr, s.split('.'), dut)
```
